run_matrix.py

```diff
@@ -112,18 +114,29 @@ def test_{interface}_interface({fixture_id}: InterfaceTester):
     {fixture_id}.configure(
         interface_name="{interface}",
         interface_version={version},
+        repo="{repo}",
+        branch="{branch}",
     )
     {fixture_id}.run()
 """
 
 
 def _generate_test(
-    interface: str, test_path: Path, fixture_id: str, version: int
+    interface: str,
+    test_path: Path,
+    fixture_id: str,
+    version: int,
+    repo: str,
+    branch: str,
 ) -> Path:
     """Generate a pytest file for a given charm and interface."""
     logging.info(f"Generating test file for {interface} at {test_path}")
     test_content = _TEST_CONTENT.format(
-        interface=interface, fixture_id=fixture_id, version=version
+        interface=interface,
+        fixture_id=fixture_id,
+        version=version,
+        repo=repo,
+        branch=branch,
```

Comment:
Probably out of scope: I'm not a fan of generating Python code in order to run tests.

I think that the same effect could be achieved by `pytest.mark.parametrize` or a plugin that defines `pytest_collection_modifyitems`.

Or, alternatively, the matrix could be a GitHub matrix, and the test runner would get the bits as arguments.

At the same time, this could very well be out of scope, as I imagine we've got to get [most of] the actual tests running first. The automation mechanics could be considered polish by comparison.